src/components/Welcomes/QRScanner/index.tsx

```diff
@@ -15,13 +19,50 @@ export default function QRScanner(
     props: Props & React.DetailedHTMLProps<React.VideoHTMLAttributes<HTMLVideoElement>, HTMLVideoElement>,
 ) {
     const { scanning, onResult, onError, ...videoProps } = props
-
     const video = useRef<HTMLVideoElement | null>(null)
 
-    useQRCodeScan(video, scanning, onResult, onError)
+    const [result, setResult] = useState('')
+
+    function onQRResult(file: string) {
+        if (video.current) {
+            video.current.pause()
+        }
+        setResult(file)
+    }
+
+    function onRescanClicked() {
+        setResult('')
+        video.current!.play()
+    }
+
+    useQRCodeScan(video, scanning, onQRResult, onError)
     return (
-        <>
+        <div style={{ position: 'relative' }}>
             <video ref={video} aria-label="QR Code scanner" {...videoProps} />
-        </>
+            {result && (
+                <div
+                    style={{
+                        background: 'rgba(255,255,255,0.3)',
```

Comment:
Does this color fits both in Maskbook light and dark theme? (You can use makeStyles to adopt current theme)

Reply:
I don't think it fits any styles. It will always cover the scanned QR code. Maybe black mask with white text is better(but I don't quite know how to change the text's color).

Reply:
I don't think it fits any styles. It will always cover the scanned QR code. Maybe black mask with white text is better(but I don't quite know how to change the text's color).